opencart 1.5.6.4/upload/catalog/controller/smailyforopencart/rss.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+class ControllerSmailyForOpencartRss extends Controller {
+
+    public function index() {
+        // Load models.
+        $this->load->model('catalog/product');
+        $this->load->model('tool/image');
+        $this->load->model('localisation/currency');
+
+        // Get params from url for filtering products.
+        if (isset($this->request->get['category'])) {
+            $category = $this->request->get['category'];
+        } else {
+            $category = null;
+        }
+        if (isset($this->request->get['limit'])) {
+            $limit = (int) $this->request->get['limit'];
+        } else {
+            $limit = 50;
+        }
+        // Xml header.
+        $this->response->addHeader('Content-Type: text/xml; charset=utf-8');
```

Comment:
Try to keep code that logically belongs together... well, together. I suggest moving this to just before line 75.